### ISSUE TITLE
feat: add consistent-emphasis-style rule (MD049)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#200]
## Overview

Implement the `consistent-emphasis-style` rule (equivalent to markdownlint MD049 / remark-lint `emphasis-marker`).

Closes part of #76.

## Problem

Markdown documents often mix asterisk (`*text*`) and underscore (`_text_`) emphasis inconsistently. This applies to both regular emphasis (`*` / `_`) and strong emphasis (`**` / `__`).

## Proposed Rule Key

`consistent-emphasis-style`

## Detection Logic

Scan each line (outside fenced code blocks and inline code spans) for emphasis spans. Depending on the configured `style`:

| Style | Behaviour |
|---|---|
| `consistent` (default) | The **first** emphasis character found in the document sets the expected style; any subsequent span using a different character is flagged |
| `asterisk` | Every emphasis span must use `*`; underscore spans are always flagged |
| `underscore` | Every emphasis span must use `_`; asterisk spans are always flagged |

Underscores inside words (e.g. `snake_case`) are not treated as emphasis.

**Invalid** (style: `consistent`, first span is asterisk):
```markdown
This is *italic* text.

This is _also italic_ text.
```

**Valid** (style: `consistent`):
```markdown
This is *italic* and **bold** text.

Another *italic* span here.
```

**Valid** (style: `underscore`):
```markdown
This is _italic_ and __bold__ text.
```

## Fix

Replace the flagged emphasis marker with the expected one. For example, if the expected style is asterisk, change `_italic_` to `*italic*` and `__bold__` to `**bold**`.

## Implementation Notes

- Rule key: `consistent-emphasis-style`
- Config field: `style` string (`"consistent"` | `"asterisk"` | `"underscore"`), default `"consistent"`
- Default enabled with `style: "consistent"`
- Skip content inside fenced code blocks and inline code spans
- Underscores inside words (`snake_case`) must not be treated as emphasis
- Reuse `stripInlineCode`, `firstNonSpaceByte`, `openingFenceMarker`, `IsClosingFence` utilities

### Config

**Default (`DefaultConfigJSON`):**
```json
"consistent-emphasis-style": { "style": "consistent" }
```

**Default (`Default()` in Go):**
```go
"consistent-emphasis-style": {
    Enabled:  true,
    Severity: SeverityError,
    Options:  map[string]interface{}{"style": "consistent"},
},
```

**User customisation examples:**
```json
"consistent-emphasis-style": { "style": "asterisk" }
```
```json
"consistent-emphasis-style": false
```

## References

- [markdownlint MD049](https://github.com/DavidAnson/markdownlint/blob/main/doc/md049.md)
- [remark-lint emphasis-marker](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-emphasis-marker)
